### PR TITLE
Add setup.py file

### LINKED ISSUE #4: Create Python wheel of application
A Python wheel allows users to build and install the package without running the code directly from source. If the binary is saved somewhere easily accessible, it can also be installed directly without requiring the repository to be cloned to a local machine first.

### PR DESCRIPTION
**Pull request by roclark**
The `setup.py` file makes it easy to install the application for other users and adds the capability of building a Python wheel which can be distributed publicly.

Closes #4 

Signed-Off-By: Robert Clark <roclark@nvidia.com>

### CODE REVIEW

**Comment by roclark** [comment]
Took a lot of thrashing to get there, but think this is finally in good shape and looking the way I hoped it would. Users can now install the package with:

```bash
pip install imageinary
```

This will only install the necessary packages for generating standard images, such as JPG and PNG. There are also options for MXNet, TFrecords, and `all` which install all of the other dependencies to unlock the remaining functionality. I felt this would be helpful to not force users to download/install the heavyweight MXNet and TensorFlow binaries if they are only going to create standard image files.

File: tests/functional/test_tfrecord.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 import os
 from click.testing import CliRunner
 from glob import glob
-from imagine import main
+from imagine.imagine import main
```
Comment (joehandzik) [inline tests/functional/test_tfrecord.py:19]: For this and the other tests, will users need to import main from imagine.imagine, or will it look more like the test_units.py code where one can import in a less repetitious way?

File: imagine/imagine.py [hunk 2]
```diff
@@ -17,11 +18,21 @@
 import click
 from PIL import Image
 from multiprocessing.pool import Pool
-from mxnet.recordio import IRHeader, MXIndexedRecordIO, pack
+try:
+    from mxnet.recordio import IRHeader, MXIndexedRecordIO, pack
+except ImportError:
+    IRHeader = None
```
Comment (joehandzik) [inline imagine/imagine.py:24]: This was my main question - how do we message that further dependencies are necessary? Looks alright to me.

File: tests/functional/test_tfrecord.py (hunk 1 above)
Comment (roclark) [reply tests/functional/test_tfrecord.py:19]: This update just adds the ability for users to be able to install the package and run it from the CLI doing something like

```bash
imagine create-images ...
```

While users _can_ import the library into their code with this update, I think we need to make some larger updates to the code and add more formal documentation to get it the way we would expect other users to be able to import. I am planning on working on that for the next update, but this doesn't support that functionality at the moment. If you think it belongs here now, I'd be happy to include it!

File: imagine/imagine.py (hunk 2 above)
Comment (roclark) [reply imagine/imagine.py:24]: Sadly, I don't think there's a great, clean way to really do this, but the way I implemented it seems to be the most commonly preferred method.

File: tests/functional/test_tfrecord.py (hunk 1 above)
Comment (joehandzik) [reply tests/functional/test_tfrecord.py:19]: Understood - nah, not needed yet, just wanted to understand how we wanted the library aspect to work. If that's a future feature and this is just test kludge for now, I can live with that. :)

File: imagine/imagine.py (hunk 2 above)
Comment (joehandzik) [reply imagine/imagine.py:24]: Ok, just so long as you did some investigation into what options exist, I'm fine with this.

**Review by joehandzik — APPROVED** [review]

File: tests/functional/test_tfrecord.py (hunk 1 above)
Comment (roclark) [reply tests/functional/test_tfrecord.py:19]: I will add that to the roadmap!

**Pull request merged**